src/main/java/com/odysseusinc/athena/service/impl/DownloadShareServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+/*
+ *
+ * Copyright 2018 Odysseus Data Services, inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Company: Odysseus Data Services, Inc.
+ * Product Owner/Architecture: Gregory Klebanov
+ * Authors: Pavel Grafkin, Vitaly Koulakov, Maria Pozhidaeva
+ * Created: April 4, 2018
+ *
+ */
+
+package com.odysseusinc.athena.service.impl;
+
+import com.google.common.base.Splitter;
+import com.odysseusinc.athena.api.v1.controller.converter.UrlBuilder;
+import com.odysseusinc.athena.api.v1.controller.dto.vocabulary.DownloadShareChangeDTO;
+import com.odysseusinc.athena.model.athena.DownloadBundle;
+import com.odysseusinc.athena.model.athena.DownloadShare;
+import com.odysseusinc.athena.model.security.AthenaUser;
+import com.odysseusinc.athena.repositories.athena.DownloadShareRepository;
+import com.odysseusinc.athena.service.DownloadShareService;
+import com.odysseusinc.athena.service.mail.VocabulariesShareSender;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class DownloadShareServiceImpl implements DownloadShareService {
```

Comment:
Does this not require to be transactional?

Reply:
fixed